CrewChiefV4/UserInterface/PropertiesForm.cs

```diff
@@ -179,5 +199,203 @@ private void properties_FormClosing(object sender, FormClosingEventArgs e)
+            if (text == DEFAULT_SEARCH_TEXT)
+            {
+                this.searchTextPrev = text;
+                return;
+            }
+
+            if (text != this.searchTextPrev)
+            {
+                // This is the case on entering the text box
+                if (string.IsNullOrWhiteSpace(text) && this.searchTextPrev != DEFAULT_SEARCH_TEXT)
+                    this.PopulatePrefsFiltered(text);
+                // General case, new filter.
+                else if (!string.IsNullOrWhiteSpace(text))
+                    this.PopulatePrefsFiltered(text);
+
+                this.searchTextPrev = text;
+            }
+        }
+
+        private void TextBox1_GotFocus(object sender, EventArgs e)
+        {
+            if (this.textBox1.Text == DEFAULT_SEARCH_TEXT)
+            {
+                this.textBox1.Text = "";
+                this.textBox1.ForeColor = Color.Black;
+            }
+        }
+
+        private void TextBox1_LostFocus(object sender, EventArgs e)
+        {
+            if (string.IsNullOrWhiteSpace(this.textBox1.Text))
+            {
+                this.textBox1.Text = DEFAULT_SEARCH_TEXT;
+                this.textBox1.ForeColor = Color.Gray;
+                this.button2.Select();
+            }
+        }
+
+        private void TextBox1_KeyDown(object sender, KeyEventArgs e)
+        {
+            if (e.KeyCode == Keys.Escape)
+            {
+                this.textBox1.Select();
+                this.textBox1.Text = "";
+                this.button2.Select();
+
+                if (!string.IsNullOrWhiteSpace(this.searchTextPrev) && this.searchTextPrev != DEFAULT_SEARCH_TEXT) 
+                    this.PopulatePrefsFiltered(null);
+            }
+            else if (e.KeyCode == Keys.Enter)
+            {
+                this.searchTextPrev = this.textBox1.Text;
+                this.PopulatePrefsFiltered(this.searchTextPrev);
+            }
+        }
+
+        private void PropertiesForm_KeyDown(object sender, KeyEventArgs e)
+        {
+            if (e.Control && e.KeyCode == Keys.E)
+                this.textBox1.Select();
+            else if (e.KeyCode == Keys.Escape)
+            {
+                // Close only if no search is active.
+                if (this.textBox1.Text == DEFAULT_SEARCH_TEXT)
+                    this.Close();
+                else
+                    this.TextBox1_KeyDown(sender, e); // Otherwise, forward.
+            }
+        }
+
+        private void PopulatePrefsFiltered(string filter)
+        {
+            this.flowLayoutPanel1.SuspendLayout();
+
+            bool anyHits = false;
+            foreach (var ctrl in this.flowLayoutPanel1.Controls)
+            {
+                if (ctrl is StringPropertyControl)
+                {
+                    var spc = ctrl as StringPropertyControl;
+                    if (string.IsNullOrWhiteSpace(filter) || spc.propertyId.ToUpperInvariant().Contains(filter.ToUpperInvariant()))
```

Comment:
can we use propertyId here? If we have non-English friendly labels the propertyId won't match. Is if possible to get the ...PropertyControl's label field, instead of its propertyId